Engine/src/Engine/SceneBuilder/Scene.h

```diff
@@ -99,14 +110,37 @@ namespace Engine
         int m_windowWidth{1920};
         int m_windowHeight{1080};
 
+        //Added this to keep track of the entities created
+        std::unordered_map<std::string, entt::entity> m_entityHandles;
+
+
+        //flag, if true: ui is rendered, if false: no ui is rendered
+        bool showUI = true;
+
         //main menu is the UI element that shows the game
         MainMenu m_mainMenu;
 
         //explorer panel is the UI element that shows the file explorer
         ExplorerPanel m_explorerPanel;
 
         //entities panel is the UI element that lists all the entities
-        EntitiesPanel m_entitiesPanel;
+        HierarchyPanel m_hierarchyPanel;
+
+        //inspector panel is the UI element that lists all the components for each entity
+        InspectorPanel m_inspectorPanel;
+
+        //game panel is the UI element that shows the render of the game
+        GamePanel m_gamePanel;
+
+        //dialog box to allow user to input the tag name for an entity
+        TagDialog m_tagDialog;
+
+        enum ComponentPanels
```

Comment:
since this is an UI specific enum, it would be better placed outside of scene.

Reply:
I moved it to ComponentsPanel.cpp